deepchem/models/callbacks.py

```diff
@@ -97,3 +98,7 @@ def __call__(self, model, step):
       # Log data to Wandb
       data = {'eval/' + k: v for k, v in scores.items()}
       model.wandb_logger.log_data(data, step, dataset_id=id(self.dataset))
+
+    # For torch models, need to set model back to train mode after eval
+    if isinstance(model, TorchModel):
+      model.model.train()
```

Comment:
Per discussion instead of lines 101-104, consider inserting at line 81:

`
if isinstance(model, TorchModel):
    with torch.no_grad():
        model.model.eval()
        scores = model.evaluate(self.dataset, self.metrics, self.transformers)
    model.model.train()
else:
    scores = model.evaluate(self.dataset, self.metrics, self.transformers)
`

Reply:
Hmm, was able to track down the `model.eval()` call. See [the _predict method of TorchModel](https://github.com/deepchem/deepchem/blob/master/deepchem/models/torch_models/torch_model.py#L574). Perhaps that's a better place for my change, as well as adding `torch.no_grad()`

Reply:
Moved this to the predict method